packages/babel-parser/src/util/scope.js

```diff
@@ -102,7 +102,7 @@ export default class ScopeHandler<IScope: Scope = Scope> {
   // > treated like var declarations rather than like lexical declarations.
   treatFunctionsAsVarInScope(scope: IScope): boolean {
     return !!(
-      scope.flags & SCOPE_FUNCTION ||
+      scope.flags & SCOPE_FUNCTION || scope.flags & SCOPE_STATIC_BLOCK ||
```

Comment:
```suggestion
      scope.flags & (SCOPE_FUNCTION | SCOPE_STATIC_BLOCK) ||
```
It can be simplified to one condition.